src/object/C4Object.cpp

```diff
@@ -225,9 +225,12 @@ void C4Object::AssignRemoval(bool exit_contents)
 	{
 		return;
 	}
+	// Set status to deleting, so that callbacks in this function that might delete
+	// the object do not delete it a second time.
+	// TODO: This causes problems if the status should change during one of those
+	// callbacks, or if the status was C4OS_INACTIVE
```

Comment:
See the additional TODO:
Either a) we keep the old "if (Status == C4OS_DELETED) return" calls to be safe and remove the C4OS_DELETING again; or
b) we make "object is being deleted" a bool that is independent from the status?

I guess the following options are not a valid solution:
c) Make the status bitwise, instead of exclusive. So you could have Status = C4OS_DELETING | C4OS_INACTIVE. This is not really safe, because I assume that status was assigned directly in other places, meaning that we have to adjust other files, too
d) Set the status C4OS_DELETED at the beginning of the function and remove the C4OS_DELETING again. This was probably not done, because the rest of the engine assumes that the object is gone when the status is C4OS_DELETED, so the callbacks might not do their stuff properly if the status is set beforehand?